source/renderer/assets/codemirror/zettlr-plugin-render-math.js

```diff
@@ -12,13 +12,12 @@
 })(function (CodeMirror) {
   'use strict'
 
-  // Matches all inlines according to the Pandoc documentation
+  // Matches all equations contained in one line according to the Pandoc documentation
   // on its tex_math_dollars-extension.
   // More information: https://pandoc.org/MANUAL.html#math
-  // First alternative is only for single-character-equations
-  // such as $x$. All others are captured by the second alternative.
-  var inlineMathRE = /(?<!\\)\${1,2}([^\s\\])\${1,2}(?!\d)|(?<!\\)\${1,2}([^\s].*?[^\s\\])\${1,2}(?!\d)/g
-  var multilineMathRE = /^\s*\$\$\s*$/
+  // First alternative is only for single-character-equations such as $x$. All others are captured by the second alternative.
+  var inlineMathRE = /(?<!\\)(?<bracket>\${1,2})(?<eq>[^\s\\])\k<bracket>(?!\d)|(?<!\\)(?<bracketTwo>\${1,2})(?<eqTwo>[^\s].*?[^\s\\])\k<bracketTwo>(?!\d)/g
```

Comment:
I didn't really change this regex (or make it more complicated per se), only required that the beginning and closing dollar signs have to be the same, i.e. `$x$` and `$$x$$` are fine but `$x$$` not. 

The side effect was that `$x$$y$` now works. I agree that it's ugly, but semantically correct. In my opinion it is more important to flag `$x$$` as wrong.